Tests/PennyTests/GatewayProcessingTests.swift

```diff
@@ -43,11 +43,57 @@ class GatewayProcessingTests: XCTestCase {
                 storage: storage
             )
         }
-        await stateManager._tests_reset()
+        await BotStateManager.shared._tests_reset()
         Task { try await Penny.main() }
-        await manager.waitUntilConnected()
+        await waitForStateManagerShutdownAndDidShutdownSignals()
     }
-    
+
+    func waitForStateManagerShutdownAndDidShutdownSignals() async {
+        /// Wait for the shutdown signal, then send a `didShutdown` signal.
+        /// in practice, the `didShutdown` signal is sent by another Penny that is online.
+        while let possibleSignal = await responseStorage.awaitResponse(
+            at: .createMessage(channelId: Constants.Channels.logs.id)
+        ).value as? Payloads.CreateMessage {
+            if possibleSignal.content?.contains(StateManagerSignal.shutdown.value) == true {
+                let content = await BotStateManager.shared._tests_didShutdownSignalEventContent()
+                await manager.send(event: .init(
+                    opcode: .dispatch,
+                    data: .messageCreate(.init(
+                        id: try! .makeFake(),
+                        channel_id: Constants.Channels.logs.id,
+                        author: .init(
+                            id: Snowflake(Constants.botId),
+                            username: "Penny",
+                            discriminator: "#0"
+                        ),
+                        content: content,
+                        timestamp: .fake,
+                        tts: false,
+                        mention_everyone: false,
+                        mention_roles: [],
+                        attachments: [],
+                        embeds: [],
+                        pinned: false,
+                        type: .default,
+                        mentions: []
+                    ))
+                ))
+                break
+            }
+        }
+
+        /// Wait to make sure the `BotStateManager` cache is already populated.
+        for idx in 1...100 {
+            try? await Task.sleep(for: .milliseconds(50))
+            if await BotStateManager.shared.isCachePopulated {
+                break
+            }
+            if idx == 100 {
+                XCTFail("BotStateManager cache was too late to populate")
+            }
+        }
```

Comment:
Ewww, polling loop... I don't have a better suggestion here short of Swift 5.9-specific features, though, beyond this simplification:

```suggestion
        for _ in 1...100 where !(await BotStateManager.shared.isCachePopulated) {
            try? await Task.sleep(for: .milliseconds(50))
        }
        XCTAssert(BotStateManager.shared.isCachePopulated, "BotStateManager cache was too late to populate")
```
(Note: I have not checked that you can actually do an await like that in a loop-conditional clause; if the compiler complains, put it in the body instead.)